sql/src/test/java/io/crate/execution/engine/collect/files/FileReadingIteratorTest.java

```diff
@@ -51,50 +52,121 @@
 import java.util.List;
 import java.util.function.Supplier;
 
+import static io.crate.execution.dsl.phases.FileUriCollectPhase.InputFormat.CSV;
+import static io.crate.execution.dsl.phases.FileUriCollectPhase.InputFormat.JSON;
 import static io.crate.testing.TestingHelpers.createReference;
 
 public class FileReadingIteratorTest extends CrateUnitTest {
 
     private InputFactory inputFactory;
     private Path tempFilePath;
+    private String fileUri;
+    private File tmpFile;
+    private static final byte[] JSON_AS_MAP_FIRST_LINE = "{\"name\": \"Arthur\", \"id\": 4, \"details\": {\"age\": 38}}".getBytes(StandardCharsets.UTF_8);
```

Comment:
please avoid using `static` in tests for constants unless the creation is very expensive. We've a leak detection and anything `static` counts towards it.

Reply:
Ok, I have removed this. 👍 So I understand this, using `static` in tests can mean that the objects are still referenced and will not be garbage collected, even if they're no longer being used?

Reply:
Static references point Objects in the heap until the corresponding class is unloaded, i.e. usually when the JVM shuts down. This is not really an issue in tests but since we have a leak detection, it avoids false-positives caused by these references.